WordPress/gutenberg
# ColorPicker: Only display detailed inputs

Related:
- https://github.com/WordPress/gutenberg/issues/37981

## What?

Removes the requirement to toggle display of detailed input fields for the ColorPicker's different color models.

_This PR only addresses the UI change not updating pasted hex values. That will be addressed separately._

## Why?

To ease the editing of color values by reducing assumed knowledge and the number of clicks required to make a change via the inputs.

Further reasoning and discussion can be found in: https://github.com/WordPress/gutenberg/issues/37981

> Users who want to input their own custom HEX values need to learn to hit that icon before they can modify the current HEX value. I've found this to be confusing personally — more often than not, I'm coming here to input a custom value, not copy one, and I have to do an extra click to make that happen.

## How?

- Removes the toggling of "detailed controls" and instead immediately renders the color inputs
- Creates a new `ColorCopyButton` component internal to the `ColorPicker`. 
    - Handles copying a string representation of the current color model's value to the clipboard as per the functionality currently on trunk when the user clicks the color value prior to toggling on detailed controls.
- Removes the now unused internal `ColorDisplay` component.

## Testing Instructions

1. Ensure the existing unit tests still pass: `npm run test-unit  packages/components/src/color-picker/test`
2. Fire up Storybook and explore the new UX via the [ColorPicker Page](http://localhost:50240/?path=/story/components-colorpicker--default)
3. In the editor, create or edit a post, select a block e.g. paragraph and test configuring various custom colors still works. 

## Screenshots or screencast 

| Before | After |
|---|---|
| <video src="https://user-images.githubusercontent.com/60436221/169749207-b6a37d26-62e1-440f-b879-22feb46c67ed.mp4" /> | <video src="https://user-images.githubusercontent.com/60436221/169749224-32d4fa59-37a5-49d6-81ee-beda61b1542c.mp4" /> |
